spring-cloud-sleuth-autoconfigure/src/test/java/org/springframework/cloud/sleuth/autoconfig/wavefront/WavefrontAutoConfigurationTests.java

```diff
@@ -0,0 +1,207 @@
+						.doesNotHaveBean(io.opentracing.Tracer.class));
+	}
+
+	@Test
+	void tracingCanBeDisabled() {
+		this.contextRunner.withPropertyValues("spring.sleuth.wavefront.enabled=false")
+				.with(wavefrontMetrics(() -> mock(WavefrontSender.class)))
+				.run((context) -> assertThat(context).doesNotHaveBean(WavefrontTracingCustomizer.class)
+						.doesNotHaveBean(io.opentracing.Tracer.class));
+	}
+
+	@Test
+	void tracingIsNotConfiguredWithNonWavefrontRegistry() {
+		this.contextRunner.with(metrics()).run((context) -> assertThat(context).doesNotHaveBean(Tracer.class));
+	}
+
+	@SuppressWarnings("ConstantConditions")
+	private WavefrontSleuthBraveSpanHandler extractSpanHandler(Tracer tracer) {
+		SpanHandler[] handlers = (SpanHandler[]) ReflectionTestUtils.getField(
+				ReflectionTestUtils.getField(ReflectionTestUtils.getField(tracer, "spanHandler"), "delegate"),
+				"handlers");
+		return (WavefrontSleuthBraveSpanHandler) handlers[1];
+	}
+
+	@SuppressWarnings("unchecked")
+	private static <T extends AbstractApplicationContextRunner<?, ?, ?>> Function<T, T> wavefrontMetrics(
+			Supplier<WavefrontSender> wavefrontSender) {
+		return (runner) -> (T) runner.withBean(WavefrontSender.class, wavefrontSender)
+				.withConfiguration(AutoConfigurations.of(WavefrontMetricsExportAutoConfiguration.class))
+				.with(metrics());
+	}
+
+	@SuppressWarnings("unchecked")
+	private static <T extends AbstractApplicationContextRunner<?, ?, ?>> Function<T, T> metrics() {
+		return (runner) -> (T) runner.withPropertyValues("management.metrics.use-global-registry=false")
+				.withConfiguration(AutoConfigurations.of(MetricsAutoConfiguration.class,
+						CompositeMeterRegistryAutoConfiguration.class));
+	}
+
+	@SuppressWarnings("unchecked")
+	private static <T extends AbstractApplicationContextRunner<?, ?, ?>> Function<T, T> sleuth() {
+		return (runner) -> (T) runner.withConfiguration(AutoConfigurations.of(BraveAutoConfiguration.class));
+	}
+
+	@Configuration
+	static class Config {
+
+		@Bean
+		ApplicationTags applicationTags(Environment environment) {
+			return createFromProperties(environment);
+		}
+
+		public ApplicationTags createFromProperties(Environment environment) {
+			String application = environment.getProperty("wavefront.application.name");
+			String service = environment.getProperty("wavefront.application.service");
+			service = (StringUtils.hasText(application)) ? service : defaultServiceName(environment);
+			ApplicationTags.Builder builder = new ApplicationTags.Builder(application, service);
+			builder.cluster(environment.getProperty("wavefront.application.cluster"));
+			builder.shard(environment.getProperty("wavefront.application.shard"));
+			return builder.build();
+		}
+
+		private String defaultServiceName(Environment environment) {
+			String applicationName = environment.getProperty("spring.application.name");
+			return (StringUtils.hasText(applicationName)) ? applicationName : "unnamed_service";
```

Comment:
Would not this be enough (there is another one in `createFromProperties` and in the next test class too)?
```suggestion
			return environment.getProperty("spring.application.name", "unnamed_service");
```